### ISSUE TITLE
Random crash while playing songs in background

### ISSUE THREAD

**Issue by Malgotra** [Malopieds/InnerTune#99]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

Play any song in background.

### Expected behavior

Play songs normally till user doesn't pause the song. No crash.

### Actual behavior

After some time while playing songs app crashes without any logs.

### Logs

Doesn't produce. App just closes.

### InnerTune version

v0.5.6

### Android version

Android 13

### Additional information

Android 13. Motorola Moto G72

**Comment by Malgotra** [comment]
I found Same bug in main Innertune issue repository.
https://github.com/z-huang/InnerTune/issues/1270

**Comment by itsmeteddyhere** [comment]
This is probably not a problem with InnerTune but rather just Android closing it down. Try disabling battery optimizations on it.

**Comment by xdpirate** [comment]
I'm experiencing this same issue on 0.5.7, but it also presents itself with the app in the foreground. It usually happens when the playback of a track ends and is proceeding to the next track in the queue.

It also seems to happen when the app is playing audio while the phone is locked, and then unlocked - the app will just crash with no error or any indications as to why. I've also experienced random crashes while downloading a playlist, especially combined with unlocking the phone to check download progress.

Battery optimizations are turned off (set to "Unrestricted" background use). Android 13, Samsung Galaxy Z Fold 3. Happens both in folded and closed states.

I haven't yet been able to reproduce on the latest debug build, but will report back once I've given it thorough enough testing.

**Comment by Malopieds** [comment]
> will report back once I've given it thorough enough testing

Thanks, that could be really useful to see if that's something from InnerTune or the OS itself, some are just [worse than other](https://dontkillmyapp.com/). I personally use grapheneOS so I've never seen this around but I'll keep looking for it!

**Comment by kaneryu** [comment]
+1 on @xdpirate 's description, that's exactly how it happens to me.

**Comment by xdpirate** [comment]
Here's a followup post after thorough testing and daily use for ~2 weeks.

The debug APK does not exhibit the same behaviour described in my previous post.

I used the GH Actions built APK for commit 8fd13b047e95b78ea844027c4fc6d940c1dec1ef.

https://github.com/Malopieds/InnerTune/actions/runs/9821011196

Everything described earlier seems to have been fixed at some point between the 0.5.7 release, and this commit :)

**Comment by Malopieds** [comment]
> Everything described earlier seems to have been fixed at some point between the 0.5.7 release, and this commit :)

Wait so you mean this is not occurring anymore ?

**Comment by xdpirate** [comment]
@Malopieds Correct! The problem seems to have gone away by the commit I tested. For me, at least.

**Comment by Malopieds** [comment]
> @Malopieds Correct! The problem seems to have gone away by the commit I tested. For me, at least.

Still now? Because in between I somehow reverted this commit x)

Is it the case for everyone ?

**Comment by xdpirate** [comment]
I have not tested any newer builds, but I can do that at some point soon.

**Comment by Malopieds** [comment]
> I have not tested any newer builds, but I can do that at some point soon.

Should be starting from 7cf22d04a9eee8863a506a405ed64d0d9baba47e

**Comment by xdpirate** [comment]
This seems to be gone from the latest F-Droid version (0.5.13), or at the very least I can't reproduce it anymore. Unless @Malgotra is still experiencing this, I think the issue can be closed.
